package/gluon-scheduled-domain-switch/luasrc/usr/bin/gluon-switch-domain

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/lua
+
+local uci = require('simple-uci').cursor()
+local unistd = require 'posix.unistd'
+local site = require 'gluon.site'
+
+-- Returns true if given domain is present on device
+function domain_exists(domain_name)
+	return unistd.access("/lib/gluon/domains/"..domain_name..".json")
+end
+
+-- Returns true if node was offline long enough to perform domain switch
+function switch_after_min_reached()
+	if not unistd.access("/tmp/gluon_gw_unreach") then
+		return false
+	end
+
+	local current_epoch = os.time()
+	local switch_after_sec = site.domain_switch.switch_after_offline_mins() * 60
+
+	local f = io.open("/tmp/gluon_gw_unreach")
+	if not f then
+		return false
+	end
+	local offline_since = f:read('*a')
+	f:close()
+
+	local offline_time_sec = current_epoch - offline_since
```

Comment:
Not a Lua expert. But could this over- or underflow in 2038 or could the automatic string to float conversion result in wrong values? (date --date="@`echo $((2**31))`" => "Tue 19 Jan 04:14:08 CET 2038".

Reply:
Lua switches to doubles when integers are insufficient (originally, Lua 5.1 did not support integers at all, and it always used floating-point math; OpenWrt backported integer support from newer Lua versions). I'm not sure of what kind of conversion issues you are thinking.

I think adding a *tonumber* around the `f:read()` would make the code a bit clearer.

I assume the current os.time() implementation will break in 2038, but that's a different story... glibc and kernel developers are working on that issue.